src/Compilers/Core/Portable/Operations/IFlowCaptureReferenceOperation.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Operations
+{
+    /// <summary>
+    /// Represents a point of use of an intermediate result captured earlier.
+    /// PROTOTYPE(dataflow): Finalize the design how capturing/referencing intermediate results is represented.
+    /// </summary>
+    public interface IFlowCaptureReferenceOperation : IOperation
+    {
+        /// <summary>
+        /// An id used to match references to the same intermediate result.
+        /// </summary>
+        int Id { get; }
```

Comment:
Should this be the `IFlowCaptureOperation` instead of an id?

Reply:
We'd have to duplicate the value being captured, Id seems correct. We should probably create an extension method to get the IFlowCaptureOperation from a Reference though.

---
In reply to: [163285768](https://github.com/dotnet/roslyn/pull/24358#discussion_r163285768) [](ancestors = 163285768)

Reply:
Duplicated to set the parent? The parent of the `IFlowCaptureOperation` should be the original parent.

Reply:
> Should this be the IFlowCaptureOperation instead of an id?

No, IFlowCaptureOperation is used at the place when result is captured, and refers to the node that represents the captured value. This node designates the place where that captured result is used later on, depending on the control flow we might reach this place from different code paths, each capturing different value in the same "slot" identified by the ```Id```.

---
In reply to: [163285768](https://github.com/dotnet/roslyn/pull/24358#discussion_r163285768) [](ancestors = 163285768)